plugins/deployment_manager/DeploymentManager.ts

```diff
@@ -132,6 +132,28 @@ export class DeploymentManager {
     );
   }
 
+  /* Deploys missing contracts from the deployment, using the user-space deploy.ts script */
+  async deployMissing(force: boolean = true) {
+    // XXX if this is idempotent we can just always deploy, and do the same for dev
+    //  as is, won't handle cases where the deploy script adds roots or partial redeploys
+    // if force or there are no roots, deploy
+    //  force will also have to change with idempotent deploy changes
+    //   its here for deploy task, which doesn't really care if roots exists or not
+    //    but we'll want another way to specify how idempotent should work
+    const roots = await this.getRoots();
+    if (force || roots.size == 0) {
+      // XXX noted above but cache is a misnomer since we have non-cache files its kind of managing
+      //  could either rename or move that functionality
+      const deployScript = this.cache.getFilePath({ rel: 'deploy.ts' });
```

Comment:
I think this is fine, do we do this with other types of files? I don't mind scens and DeploymentManager depending on one another, so long as they stay as plugins that _could_ be removed from Comet, which I think this preserves.

Reply:
I think its mainly `deploy.ts`, `configuration.json`, and `migrations` which scens currently need to know about